sql/analyzer/stored_procedures.go

```diff
@@ -91,6 +79,32 @@ func loadStoredProcedures(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scop
 	return scope, nil
 }
 
+// validates the plan.CreateProcedure and returns a plan.Procedure
+func validate(ctx *sql.Context, a *Analyzer, cp *plan.CreateProcedure, scope *Scope, sel RuleSelector) (*plan.Procedure, error) {
```

Comment:
(minor) Since Go doesn't overload on function args, and this `validate` function will be visible in the whole `analyzer` package, you might want to pick a slightly more specific name since the namespacing is so broad here. There's also a `validateStoredProcedure` func already that is being called by this method, so maybe something like `analyzeCreateProcedure` would work well?

Reply:
I swear I'll remember this fact one of these days! :)
Renaming to `analyzeCreateProcedure`.